settlement/src/lib.rs

```diff
@@ -0,0 +1,72 @@
+pub mod execution;
+
+use execution::*;
+use eyre::Error;
+use merkle_tree::*;
+use rust_decimal::Decimal;
+use serde::{Deserialize, Serialize};
+use simperby_common::*;
+
+/// An abstract information about a block from a settlement chain.
+#[derive(PartialEq, Eq, Debug, Serialize, Deserialize, Clone)]
+pub struct SettlementChainBlock {
+    /// The height of the block.
+    pub height: u64,
+    /// The UNIX timestamp of the block in seconds.
+    pub timestamp: u64,
+}
+
+/// An abstraction of the residential chain with its treasury deployed on it.
+///
+/// One trivial implementation of this trait would carry the API endpoint of the full node and
+/// the relayer account used to submit message delivering transactions.
+#[async_trait::async_trait]
+pub trait SettlementChain: Send + Sync {
+    /// Returns the name of the chain.
+    async fn get_chain_name(&self) -> String;
+
+    /// Checks whether the chain is healthy and the full node is running.
+    async fn check_connection(&self) -> Result<(), Error>;
+
+    /// Getes the latest finalized block on the chain.
+    async fn get_last_block(&self) -> Result<SettlementChainBlock, Error>;
+
+    /// Returns the address and the current balance (which is used to pay the gas fee) of the relayer account in this chain.
+    ///
+    /// Note that there is no authority of the relayer account; it is just a convenient account to pay the gas fee
+    /// (i.e., there is no need to check the transaction submitter by the contract).
+    async fn get_relayer_account_info(&self) -> Result<(String, Decimal), Error>;
+
+    /// Returns the latest header that the light client has verified.
+    async fn get_light_client_header(&self) -> Result<BlockHeader, Error>;
+
+    /// Returns the current balance of a particular fungible token in the treasury contract.
+    async fn get_treasury_fungible_token_balance(&self, address: String) -> Result<Decimal, Error>;
+
+    /// Returns the current balance of a particular non-fungible token collection in the treasury contract,
+    /// identified as their token indices.
+    async fn get_treasury_non_fungible_token_balance(
+        &self,
+        address: Vec<String>,
+    ) -> Result<String, Error>;
```

Comment:
```suggestion
    ) -> Result<Decimal, Error>;
```

Reply:
It can be `Vec<Decimal>` or `Vec<u256>` if the return value is id token_ids which the address has.

Reply:
I put `Vec<String>`. Let's revisit the types later.

Reply:
#295